### ISSUE TITLE
Define implicit operator of operator precedence parser

### ISSUE THREAD

**Issue by bert2** [stephan-tolksdorf/fparsec#25]
Is it possible to add an implicit operator to an `OperatorPrecenceParser`? For instance, could the [OOP sample](https://github.com/stephan-tolksdorf/fparsec/blob/dc06e8b1ece3fac7f92f0d556ae164e6b2a38cfe/Samples/Calculator/calculator.fs) be extended such that the test case `calculate "(1+1)(2+2)" |> equals 8.` (implicit multiplication) does not fail?

As far as I can see the OOP type does not support this. In case that's true: would you consider this feature a worthwhile extension of the type?

**Comment by bert2** [comment]
I created [a fork](https://github.com/bert2/fparsec) where I added support for the simple case that an implicit operator can be derived from an existing explicit one (e.g. implicit multiplication in the calculator sample). The solution is kinda ugly, but the original code is not exactly pretty either.

What I'm still working on are implicit operators that have no explicit operator symbol (e.g. concatenation in regular expressions).
